subprojects/base-services/src/main/java/org/gradle/internal/classloader/ClassLoaderUtils.java

```diff
@@ -51,4 +91,16 @@ public static void disableUrlConnectionCaching() {
             throw UncheckedException.throwAsUncheckedException(e);
         }
     }
+
+    public static JavaMethod<ClassLoader, Package[]> getPackagesMethod() {
+        return GET_PACKAGES_METHOD;
+    }
+
+    public static JavaMethod<ClassLoader, Package> getPackageMethod() {
+        return GET_PACKAGE_METHOD;
+    }
+
+    public static <T> Class<T> define(ClassLoader targetClassLoader, String className, byte[] clazzBytes) {
+        return Cast.uncheckedCast(UNSAFE.defineClass(className, clazzBytes, 0, clazzBytes.length, targetClassLoader, null));
```

Comment:
Make sure you run a full CI pipeline before merging this, to check this on all the various JVM versions.